mlflow/utils/thread_utils.py

```diff
@@ -0,0 +1,36 @@
+import os
+import threading
+
+_thread_locals = threading.local()
+_globals = {}
+_globals_lock = threading.Lock()
+
+
+def set_thread_local_var(key, value):
```

Comment:
Why do we need to set this dynamically?

Reply:
this is a utility function to set thread local variable value. we need to change  thread local variable sometimes